tests/nonbonded/test_consistency.py

```diff
@@ -0,0 +1,121 @@
+    example_nonbonded_params,
+    example_nonbonded_exclusion_idxs,
+    example_nonbonded_exclusion_scales,
+    example_conf,
+    example_box,
+    rng: np.random.Generator,
+):
+    conf = example_conf[:num_atoms, :]
+    params_initial = example_nonbonded_params[:num_atoms, :]
+    params = gen_params(params_initial, rng) if interpolated else params_initial
+
+    exclusion_idxs, exclusion_scales = filter_valid_exclusions(
+        num_atoms, example_nonbonded_exclusion_idxs, example_nonbonded_exclusion_scales
+    )
+
+    lambda_plane_idxs = rng.integers(-2, 3, size=(num_atoms,), dtype=np.int32)
+    lambda_offset_idxs = rng.integers(-2, 3, size=(num_atoms,), dtype=np.int32)
+
+    ligand_idxs = rng.choice(num_atoms, size=(num_atoms_ligand,), replace=False).astype(np.int32)
+    host_idxs = np.setdiff1d(np.arange(num_atoms), ligand_idxs).astype(np.int32)
+
+    make_ref_potential = NonbondedInterpolated if interpolated else Nonbonded
+    ref_impl = make_ref_potential(
+        exclusion_idxs, exclusion_scales, lambda_plane_idxs, lambda_offset_idxs, beta, cutoff
+    ).unbound_impl(precision)
+
+    if disable_hilbert_sort:
+        ref_impl.disable_hilbert_sort()
+
+    def make_allpairs_potential(atom_idxs):
+        make_potential = NonbondedAllPairsInterpolated if interpolated else NonbondedAllPairs
+        return make_potential(
+            lambda_plane_idxs,
+            lambda_offset_idxs,
+            beta,
+            cutoff,
+            atom_idxs,
+            disable_hilbert_sort=disable_hilbert_sort,
+        )
+
+    def make_ixngroup_potential(ligand_idxs):
+        make_potential = NonbondedInteractionGroupInterpolated if interpolated else NonbondedInteractionGroup
+        return make_potential(
+            ligand_idxs,
+            lambda_plane_idxs,
+            lambda_offset_idxs,
+            beta,
+            cutoff,
+            disable_hilbert_sort=disable_hilbert_sort,
+        )
+
+    def make_pairlist_potential(exclusion_idxs, exclusion_scales):
+        make_potential = NonbondedPairListNegatedInterpolated if interpolated else NonbondedPairListNegated
+        return make_potential(exclusion_idxs, exclusion_scales, lambda_plane_idxs, lambda_offset_idxs, beta, cutoff)
+
+    test_impl = FanoutSummedPotential(
+        [
+            make_allpairs_potential(host_idxs),
+            make_allpairs_potential(ligand_idxs),
+            make_ixngroup_potential(ligand_idxs),
+            make_pairlist_potential(exclusion_idxs, exclusion_scales),
+        ]
+    ).unbound_impl(precision)
+
+    du_dx_ref, du_dp_ref, du_dl_ref, u_ref = ref_impl.execute(conf, params, example_box, lamb)
+    du_dx_test, du_dp_test, du_dl_test, u_test = test_impl.execute(conf, params, example_box, lamb)
+
+    np.testing.assert_array_equal(du_dx_test, du_dx_ref)
+
+    if interpolated:
+        # TODO: why does interpolation break bitwise equivalence?
+        np.testing.assert_allclose(du_dp_test, du_dp_ref, rtol=1e-10, atol=1e-10)
```

Comment:
Not completely sure why this is necessary yet. At least all the other outputs seem bitwise equivalent (and `du_dp` for the non-interpolated case)

Reply:
Resolved offline in conversation with @proteneer. Bitwise equivalence is not possible with the current structure of the code, ultimately due to the fact that the distributive property of multiplication breaks bitwise equivalence (even when summation is done in fixed point), i.e. `c * fixed_sum(a, b) != fixed_sum(c * a, c * b)` (see example in [gist](https://gist.github.com/mcwitt/904adca81a9a67dd707c941fdbf6a1af)).

Added a note in 15a17b7.